### PR TITLE
[INFINITY-2837] readiness should be checked at startup

### PR DESCRIPTION
**Pull request by elezar**
This is another attempt at #1966.

The readiness check should be used to determine whether a task is at its RUNNING goal state.

If this is not taken into account, a scheduler restart would result in a RUNNING task with a failed readiness check being marked as completed.

Ideally we should use similar logic for task updates and the initial determination of the steps.

Due to the fact that we don't have reliable readiness check information on 1.9 clusters, the default behaviour for a `CustomExecutor` (used as a proxy for 1.9 clusters) is the current behaviour. This should address the failing tests on 1.9 in the previous PR.

### CODE REVIEW

File: frameworks/helloworld/src/test/java/com/mesosphere/sdk/helloworld/scheduler/ServiceTest.java [hunk 1]
```diff
@@ -344,6 +344,150 @@ private void testWorldDecommission(boolean useDefaultExecutor) throws Exception
         runner.run(ticks);
     }
 
+
+    @Test
```
Comment (nickbp) [inline frameworks/helloworld/src/test/java/com/mesosphere/sdk/helloworld/scheduler/ServiceTest.java]: It looks like the main differences between these tests are:
- The readiness check exit code
- The resulting status
- Whether to use custom or default executor

Given that, could break the test out into a separate parameterized function and have these three call into that function?

File: frameworks/helloworld/src/test/java/com/mesosphere/sdk/helloworld/scheduler/ServiceTest.java [hunk 2]
```diff
@@ -344,6 +344,150 @@ private void testWorldDecommission(boolean useDefaultExecutor) throws Exception
...
+        ServiceTestRunner runner = new ServiceTestRunner();
+        ServiceTestResult result = runner.run(ticks);
+
+        // Start a new scheduler:
+        ticks.clear();
+
+        ticks.add(Send.register());
+        ticks.add(Expect.reconciledExplicitly(result.getPersister()));
+
+        // Since the readiness check of the task did not pass, we expect it to remain in the PENDING state
```
Comment (nickbp) [inline frameworks/helloworld/src/test/java/com/mesosphere/sdk/helloworld/scheduler/ServiceTest.java]: Comment is out of date

File: frameworks/helloworld/src/test/java/com/mesosphere/sdk/helloworld/scheduler/ServiceTest.java [hunk 3]
```diff
@@ -344,6 +344,150 @@ private void testWorldDecommission(boolean useDefaultExecutor) throws Exception
...
+        ServiceTestRunner runner = new ServiceTestRunner().setUseCustomExecutor();
+        ServiceTestResult result = runner.run(ticks);
+
+        // Start a new scheduler:
+        ticks.clear();
+
+        ticks.add(Send.register());
+        ticks.add(Expect.reconciledExplicitly(result.getPersister()));
+
+        // Since the readiness check of the task did not pass, we expect it to remain in the PENDING state
```
Comment (nickbp) [inline frameworks/helloworld/src/test/java/com/mesosphere/sdk/helloworld/scheduler/ServiceTest.java]: Comment is out of date

**Comment by elezar** [comment]
Thanks for your comments @nickbp. I have moved these tests to their own test class, and addressed your comments w.r.t the duplication.

**Pull request merged**